augur/dates/__init__.py

```diff
@@ -121,6 +121,12 @@ def get_numerical_date_from_value(value, fmt=None, min_max_year=None):
         except InvalidDate as error:
             raise AugurError(str(error)) from error
         return [treetime.utils.numeric_date(d) for d in ambig_date]
+    if value.startswith('[') and value.endswith(']') and len(value[1:-1].split(':'))==2:
+        # Treetime ambiguous date format, e.g. [2019.5:2020.5]
+        try:
+            return [float(x) for x in value[1:-1].split(':')]
+        except ValueError as error:
+            raise AugurError(str(error)) from error
```

Comment:
Suggestion: customize the error message.

When given something like `[2023-07-31:2023-08-06]`, the `ValueError` message looks something like:

```
could not convert string to float: '2023-07-31'
```

It's obvious what this means to us as developers, but maybe not so obvious to users. Something like this might be more clear:

```
The date range format must be `[A:B]` where both `A` and `B` are numerical dates.
```